src/components/CSActionButtonGroup/CSActionButtonGroup.component.jsx

```diff
@@ -0,0 +1,44 @@
+import React from "react";
+import { Button, Space } from "antd";
+
+import BookingStatusConstants from "../../constants/BookingStatusConstants";
+import { RoleConstant } from "../../constants/RoleConstants";
+const CSActionButtonGroupComponent = ({
+  role,
+  bookingData,
+  handleAssign,
+  rejectBooking,
+  checkOutHandler,
+  checkInHandler,
+}) => {
+  return (
+    <>
+      <Space>
+        {bookingData?.status === BookingStatusConstants.PENDING &&
+        role === RoleConstant.CUSTOMER_SERVICE ? (
+          <Button onClick={() => handleAssign({ isAccepted: true })}>
+            Process this booking
+          </Button>
+        ) : null}
+        {role === RoleConstant.CUSTOMER_SERVICE &&
+        bookingData?.status === BookingStatusConstants.REQUEST ? (
+          <Button onClick={checkInHandler}>Check in</Button>
+        ) : null}
+        {bookingData?.status === BookingStatusConstants.REQUEST ||
+        (bookingData?.status === BookingStatusConstants.PENDING &&
+          role === RoleConstant.CUSTOMER_SERVICE) ? (
```

Comment:
why do i need to check role in here ? tbh it's CS button group so if the current role is not CS then auto null ?